### PR TITLE
chore: add example of rerunning failed

### PR DESCRIPTION
**Pull request by twk3**
- Updates @currents/playwright (for additional reporter support)
- Adds an example workflow that shows how one can collect failed tests, and feed them into new runs

### CODE REVIEW

File: .github/workflows/test-basic-pwc-failed-only.yml [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+      # Also install jq here, as we use it to parse our failures out of our summary file
+      - name: Install dependencies
+        if: ${{ success() && steps.download-failures.outputs.hasFailures == 'true' }}
+        run: |
+          npm ci
+          apt update
+          apt install -y jq
+          npx playwright install chrome
+          npx playwright install
```
Comment (agoldis) [inline .github/workflows/test-basic-pwc-failed-only.yml:60]: redundant

File: .github/workflows/test-basic-pwc-failed-only.yml [hunk 2]
```diff
@@ -0,0 +1,157 @@
+# Demonstrates a workflow that initially runs the previously failed tests
+## Consists of 3 jobs: previousFailedFirst, fullRun, recordFailedTests
+##
+## previousFailedFirst - downloads previous failures for the branch, if there are any, it runs those failures using a matrix
```
Comment (agoldis) [inline .github/workflows/test-basic-pwc-failed-only.yml:4]: How does it know what is "previous"?

Comment (twk3) [reply .github/workflows/test-basic-pwc-failed-only.yml:4]: It uses the github action `dawidd6/action-download-artifact@v6`  which queries the github workflows api. It has a bunch of settings, I do see that I need to update `workflow_conclusion` to `completed`. Then what it does is look at the last completed workflow for the source branch (can be set to a commit sha, or a PR number instead, or a run_id if someone happens to want to reference a specific run ) and downloads the artifacts from that.

**Review by agoldis — COMMENTED** [review]
Trying to map this implementation to using Currents API, or enhancing `pwc` to automate the steps:
- `previousFailedFirst` - send an API call to Currents and fetch **previous** failures, the biggest challenge (IMO) is properly specifying the "previous"
- parsing the response and outputting it in a format accepted by playwright
- running playwright with only the failed items

For example
```sh
FAILED = $(pwc get-failed  --key XXX --project-id YYY) # sends and API request to Currents and parses the response 
npx pwc --shard=${{ matrix.shard }}/${{ strategy.job-total }} -- $FAILED
# or
playwright test --shard=${{ matrix.shard }}/${{ strategy.job-total }} $FAILED
```

Or even like this:
```sh
npx pwc --shard=${{ matrix.shard }}/${{ strategy.job-total }} --pwc-failed-from <ref>
```
if `--pwc-failed-from <ref>` is specified, it will send request to API to get the failures.

P.S.
Can be tricky to combine with extra tags and filters that user can provide.

File: .github/workflows/test-basic-pwc-failed-only.yml (hunk 2 above)
Comment (agoldis) [reply .github/workflows/test-basic-pwc-failed-only.yml:4]: gotcha, i think it'd be helpful to elaborate on this point.

Comment (twk3) [reply .github/workflows/test-basic-pwc-failed-only.yml:5]: Update the comment with more details

**Review by agoldis — APPROVED** [review]

**Comment by twk3** [comment]
> Trying to map this implementation to using Currents API, or enhancing pwc to automate the steps:

I like the idea of  `--pwc-failed-from` . One of the reasons I avoid the API to begin with was I wanted to see what the use-case / workflows looks like. 

Like, what does this achieve that orchestration + fail fast cannot? Couldn't we just have orchestration run failed tests first as an option instead?

**Comment by agoldis** [comment]
> > Trying to map this implementation to using Currents API, or enhancing pwc to automate the steps:
> 
> I like the idea of `--pwc-failed-from` . One of the reasons I avoid the API to begin with was I wanted to see what the use-case / workflows looks like.

I think it can only be used with `pwc` as a CLI and not as a reporter - i.e. if we ever going to do that, we'll need a "standalone" solution.



> Like, what does this achieve that orchestration + fail fast cannot? Couldn't we just have orchestration run failed tests first as an option instead?

or8n + fail fast will run all the tests and can cause a failure in different test / spec. From my experience, they know their tests are flaky and have the mindset of "each test should pass at least once" to merge.  Practically, it can be tricky to comprehend and setup the whole "orchestration" bundle. My guess is that they just want to enable the native GHA "rerun-failed" functionality. 

I know that Stephan from Ramp (laid off recently) was working on it extensively.

**Pull request merged**